src/Avalonia.Controls/Button.cs

```diff
@@ -289,8 +289,9 @@ protected override void OnKeyDown(KeyEventArgs e)
                     OnClick();
                     e.Handled = true;
                     break;
-
                 case Key.Space:
+                    // Handle Space only current focus control
```

Comment:
Nit: if possible, use a slightly better comment that explains *why* we're doing this.
For example "_Avoid handling Space if the button isn't focused: a child TextBox might need it for text input_"